### PR TITLE
role configuration update

### PR DESCRIPTION
**Pull request by abirelhalimi**

### CODE REVIEW

**Comment by abirelhalimi** [comment]
I tried doing it the other way around but it didn't work. When i annotate CrudController by @Secured("ADMIN") it stays secured even though i overrode it that's why i did the inverse. I'll try adding other controllers for public operations.

**Comment by daychissaf** [comment]
In this case you can make all Crud endpoints secured("Admin") and add separate endPoints for public, as we will separate backoffice from other operations anyway
so for user registration, you can add registration endPoint and for project list also you can add new endpoint
For Admin we have a list of project but for public we can customize this list that why please separate them

**Comment by abirelhalimi** [comment]
I created a public controller for user side operations (registration, project list, updating account..)

File: donation-web/src/main/java/com/donation/conf/SecurityConfiguration.java [hunk 1]
```diff
@@ -27,8 +27,9 @@ protected void configure(HttpSecurity http) throws Exception {
       http.cors().and()
             .csrf().disable()
             .authorizeRequests()
-            .antMatchers(HttpMethod.POST,"/api/users").permitAll()
-            .antMatchers(HttpMethod.GET,"/api/projects").permitAll()
+            .antMatchers("/signup").permitAll()
+            .antMatchers(HttpMethod.GET,"/projects").permitAll()
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/conf/SecurityConfiguration.java]: Duplicate line, I think you need to allow registration here

File: donation-web/src/main/java/com/donation/web/crud/PublicController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import com.donation.crud.ProjectCrudService;
+import com.donation.crud.UserCrudService;
+import com.donation.donor.model.Project;
+import com.donation.donor.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/web/crud/PublicController.java]: the url prefix /api

**Comment by abirelhalimi** [comment]
This way?

File: donation-web/src/main/java/com/donation/web/crud/PublicController.java [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import org.springframework.web.bind.annotation.*;
 import java.util.List;
 
+@RequestMapping(value = "/api/")
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/web/crud/PublicController.java]: @RequestMapping(value = "/api")

File: donation-web/src/main/java/com/donation/web/crud/PublicController.java [hunk 4]
```diff
@@ -19,33 +20,33 @@
     ProjectCrudService projectCrudService;
 
     //Everybody can create an account
-    @PostMapping("/api/signup")
+    @PostMapping("signup")
     public User signUp(@RequestBody User user) {
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/web/crud/PublicController.java]: please move registration domain to separate Controller and Rename this one

File: donation-web/src/main/java/com/donation/web/crud/AssociationCrudController.java [hunk 5]
```diff
@@ -3,11 +3,16 @@
 import com.donation.common.CrudService;
 import com.donation.crud.AssociationCrudService;
 import com.donation.donor.model.Association;
+import com.donation.donor.model.Project;
 import com.donation.web.common.CrudController;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/web/crud/AssociationCrudController.java]: Unused imports

File: donation-web/src/main/java/com/donation/web/crud/ProjectCrudController.java [hunk 6]
```diff
@@ -5,9 +5,13 @@
 import com.donation.donor.model.Project;
 import com.donation.web.common.CrudController;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/web/crud/ProjectCrudController.java]: unused imports

File: donation-web/src/main/java/com/donation/web/crud/SponsorCrudController.java [hunk 7]
```diff
@@ -2,12 +2,17 @@
...
 import com.donation.common.CrudService;
 import com.donation.crud.SponsorCrudService;
+import com.donation.donor.model.Association;
 import com.donation.donor.model.Sponsor;
 import com.donation.web.common.CrudController;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/web/crud/SponsorCrudController.java]: remove unused import

File: donation-web/src/main/java/com/donation/web/crud/UserCrudController.java [hunk 8]
```diff
@@ -5,9 +5,13 @@
 import com.donation.donor.model.User;
 import com.donation.web.common.CrudController;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/web/crud/UserCrudController.java]: same here

File: donation-web/src/main/java/com/donation/web/crud/VideoCrudController.java [hunk 9]
```diff
@@ -2,12 +2,17 @@
...
 import com.donation.common.CrudService;
 import com.donation.crud.VideoCrudService;
+import com.donation.donor.model.Sponsor;
 import com.donation.donor.model.Video;
 import com.donation.web.common.CrudController;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
```
Comment (daychissaf) [inline donation-web/src/main/java/com/donation/web/crud/VideoCrudController.java]: same

**Pull request merged**